partiql-ast/src/main/resources/partiql_ast.ion

```diff
@@ -196,10 +196,29 @@ type::[
   timestamp_with_tz::{ precision: optional::int },             // TIMESTAMP [(<int>)] WITH TIMEZONE
   interval::{ precision: optional::int },                      // INTERVAL
   bag::{},                                                     // BAG
+  // Keep the list type for backward compatible reason
+  // user should migrate to the array type
   list::{},                                                    // LIST
+  array::{ type: optional::type },                             // ARRAY
   sexp::{},                                                    // SEXP
+  // Keep the tuple type for backward compatible reason
+  // migration: in v0.15.0: struct modeling should NOT be Introduced,
+  //  users of struct type should be migrate to tuple type.
+  //  in v1.0.0: introduce the struct modeling change, all users from tuple type should migrate to struct type.
```

Comment:
Comment is a little confusing. If I understand correctly,
1. `struct` is added in v0.15.0
2. `tuple` is kept in v0.15.0 but "deprecated"; users are advised to use `struct`
3. `tuple` is removed in v1.0.0 and users have to use `struct`?

Reply:
Tuple/Struct is a bit more difficult, let's first talk about Array/List

Consider: 

```
// in 0.14.x, we only have typeList
// LIST get routed to typeList()
typeList()

// In 0.15.x 
@Deprecated, migrate to new APIs 
typeList()
typeArray(elementType = ...)

// in 1.0.x 
typeArray(elementType = ...)
```
The List/Array migration is pretty clean due to the added typeArray()

Consider the Tuple/Struct Migration.... We could do something similar, but since this PR targets the 1.0, I just directly modified the API. 

The comment purposed a different migration strategy: 
```
// v0.14.x 
typeStruct()
typeTuple() 

// v0.15.x
@Deprecated, move the usage to typeTuple()
typeStruct()

// v0.16.x -- intermediate to avoid confusion, we don't really need this
typeTuple()

// v0.17.x
@Deprecated
typeTuple()
typeStruct(fields = ....)

// v1.0.x 
typeStruct(fields = ...)
```

The goal for this PR is to demonstrate the APIs (`typeStruct` and `typeArray`) in v1 release, without actually deleting other APIs.

Reply:
The deprecation plan for ARRAY/LIST makes sense to me.

---

Followup to the offline discussion, I think condensing the changes across fewer major version bumps can be easier to understand. Consider:

```
// v0.14.x -- the current state in `main`
typeTuple()

// v0.15.x -- the next release
@Deprecated // recommend using `typeStruct`
typeTuple()
typeStruct()

// v1.0.x
typeStruct() // `typeTuple` now deleted
```

Reply:
Whatever release process we choose though, I think this comment in the `partiql_ast.ion` file should be updated.